src/domain/leads/leads.repo.ts

```diff
@@ -0,0 +1,26 @@
+import { leadModel } from "./leads.model";
+import { CreateLeadDto, LeadDto } from "./dto/leads.dto";
+
+class leadRepo {
+  public leadModel = leadModel;
+
+  create = async (data: CreateLeadDto): Promise<CreateLeadDto> => {
+    return this.leadModel.create(data);
+  };
+
+
+  getAll = async (): Promise<LeadDto[]> => {
```

Comment:
bu joylarda interface bilan check qilsa boladi hammasida

Reply:
Repodagi **getAll** va **create** funksiyalari interface bilan tekshiriladigan qilindi:
`create = async (data: Ilead): Promise<Ilead> => {
        return this.leadModel.create(data);
    };
`
`
    getAll = async (): Promise<Ilead[]> => {
        return this.leadModel.find().lean();
    };
`